pkg/registry/apis/dashboard/schema_validation.go

```diff
@@ -0,0 +1,69 @@
+package dashboard
+
+import (
+	"context"
+	_ "embed"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v0alpha1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1alpha1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v2alpha1"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ValidateDashboardSpec validates the dashboard spec and throws a detailed error if there are validation errors.
+func ValidateDashboardSpec(ctx context.Context, obj runtime.Object, a admission.Attributes) error {
+	mode := getFieldValidationMode(a)
+	errors := ValidateDashboardSpecByMode(ctx, mode, obj)
+	if len(errors) > 0 {
+		return dashboards.NewDashboardSpecValidationErr(errors)
```

Comment:
Problem, we should return k8s error here, because client-side tooling expects a certain format.

Reply:
Got it, thanks for the context!